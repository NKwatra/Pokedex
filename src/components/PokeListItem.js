import React from 'react';
import {StyleSheet, View} from 'react-native';
import PokeCard, {LINEAR_CARD_HEIGHT, MARGIN} from './PokeCard';

export const LINEAR_ITEM_HEIGHT = 2 * LINEAR_CARD_HEIGHT + 4 * MARGIN;

export default function PokeListItem(props) {
  return (
    <View style={props.grid ? styles.grid : {}}>
      {props.items.map((item) => (
        <PokeCard {...item} grid={props.grid} key={item.name} />
      ))}
    </View>
  );
}

const styles = StyleSheet.create({
  grid: {
    flexDirection: 'row',
  },
});
