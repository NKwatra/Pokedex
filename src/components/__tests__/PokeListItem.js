import React from 'react';
import {render} from '@testing-library/react-native';
import PokeListItem from '../PokeListItem';

test('PokeListItem renders correctly in list view', () => {
  const items = [
    {name: 'A', url: 'abc'},
    {name: 'B', url: 'bcd'},
    {name: 'C', url: 'cde'},
    {name: 'D', url: 'def'},
    {name: 'E', url: 'efg'},
    {name: 'F', url: 'fgh'},
  ];
  const {toJSON} = render(<PokeListItem items={items} grid={false} />);
  expect(toJSON()).toMatchSnapshot();
});

test('PokeListItem renders correctly in grid view', () => {
  const items = [
    {name: 'A', url: 'abc'},
    {name: 'B', url: 'bcd'},
    {name: 'C', url: 'cde'},
    {name: 'D', url: 'def'},
    {name: 'E', url: 'efg'},
    {name: 'F', url: 'fgh'},
  ];
  const {toJSON} = render(<PokeListItem items={items} grid={true} />);
  expect(toJSON()).toMatchSnapshot();
});

test('given PokeListItem in list view, when user switches to Grid view, url disappears', () => {
  const items = [
    {name: 'A', url: 'abc'},
    {name: 'B', url: 'bcd'},
  ];
  const {update, getByText, queryByText} = render(
    <PokeListItem items={items} grid={false} />,
  );

  // url is shown in list view
  getByText('abc');
  getByText('bcd');

  // no url after update
  update(<PokeListItem items={items} grid={true} />);
  expect(queryByText('abc')).toBeNull();
  expect(queryByText('bcd')).toBeNull();
});
