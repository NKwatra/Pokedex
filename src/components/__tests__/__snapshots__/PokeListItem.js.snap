// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeListItem renders correctly in grid view 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "minHeight": 152,
          "paddingBottom": 4,
          "paddingTop": 0,
        },
        Object {
          "height": 423.3333333333333,
          "width": 343,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": "80%",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        A
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "minHeight": 152,
          "paddingBottom": 4,
          "paddingTop": 0,
        },
        Object {
          "height": 423.3333333333333,
          "width": 343,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": "80%",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        B
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "minHeight": 152,
          "paddingBottom": 4,
          "paddingTop": 0,
        },
        Object {
          "height": 423.3333333333333,
          "width": 343,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": "80%",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        C
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "minHeight": 152,
          "paddingBottom": 4,
          "paddingTop": 0,
        },
        Object {
          "height": 423.3333333333333,
          "width": 343,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": "80%",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        D
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "minHeight": 152,
          "paddingBottom": 4,
          "paddingTop": 0,
        },
        Object {
          "height": 423.3333333333333,
          "width": 343,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": "80%",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        E
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "minHeight": 152,
          "paddingBottom": 4,
          "paddingTop": 0,
        },
        Object {
          "height": 423.3333333333333,
          "width": 343,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": "80%",
          "marginBottom": 8,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        F
      </Text>
    </View>
  </View>
</View>
`;

exports[`PokeListItem renders correctly in list view 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        A
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        abc
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        B
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        bcd
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        C
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        cde
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        D
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        def
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        E
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        efg
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 20,
          "elevation": 2,
          "marginHorizontal": 16,
          "marginVertical": 16,
          "overflow": "hidden",
          "padding": 16,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 5,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/charmander.png",
        }
      }
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
            "textTransform": "capitalize",
          }
        }
      >
        F
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        fgh
      </Text>
    </View>
  </View>
</View>
`;
